src/python/pants/backend/jvm/tasks/scalafmt.py

```diff
@@ -26,6 +29,10 @@ def register_options(cls, register):
     super(ScalaFmt, cls).register_options(register)
     register('--configuration', advanced=True, type=file_option, fingerprint=True,
               help='Path to scalafmt config file, if not specified default scalafmt config used')
+    register('--output-dir', advanced=True, type=dir_option, fingerprint=True,
```

Comment:
While this will work for either subclass, this option only makes sense for the `ScalaFmtFormat` and `ScalaFixFix` subclasses, which actually rewrite things (and are the tasks installed in `fmt`).

So a few semi-orthogonal things:

1. I think that you could make registering the option conditional on `sideeffecting = True`, and then check it indirectly.
2. You could move the option to `RewriteBase`, and execute the file copying before `invoke_tool`... that would net you support `google-java-format` as well, which also subclasses RewriteBase.

Reply:
Did 2 :)